android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java

```diff
@@ -202,28 +203,66 @@ public Object getSelectionKey()
 				}
 			},
 			StorageStrategy.createLongStorage()
-		)
-			.withSelectionPredicate(SelectionPredicates.<Long>createSelectSingleAnything())
-			.build();
-		this.tracker.addObserver(new SelectionTracker.SelectionObserver()
-		{
+		);
 
-			@Override
-			public void onSelectionChanged()
-			{
-				super.onSelectionChanged();
+		if (proxy.getProperties().optBoolean(TiC.PROPERTY_EDITING, false)
+			&& proxy.getProperties().optBoolean(TiC.PROPERTY_ALLOWS_SELECTION_DURING_EDITING, false)) {
+			if (proxy.getProperties().optBoolean(TiC.PROPERTY_ALLOWS_MULTIPLE_SELECTION_DURING_EDITING, false)) {
+				this.tracker = trackerBuilder.withSelectionPredicate(SelectionPredicates.createSelectAnything())
+					.build();
+			} else {
+				this.tracker = trackerBuilder.withSelectionPredicate(SelectionPredicates.createSelectSingleAnything())
+					.build();
+			}
+
+			if (this.tracker != null) {
+				this.tracker.addObserver(new SelectionTracker.SelectionObserver()
+				{
+					int selectionCount = 0;
+
+					@Override
+					public void onSelectionChanged()
+					{
+						super.onSelectionChanged();
+
+						if (tracker.hasSelection() && selectionCount != tracker.getSelection().size()) {
+							selectedItems = new ArrayList<>(selectionCount);
+							final Iterator<ListItemProxy> i = tracker.getSelection().iterator();
 
-				/*if (tracker.hasSelection()) {
-					final Iterator<ListItemProxy> i = tracker.getSelection().iterator();
-					while (i.hasNext()) {
-						final ListItemProxy proxy = i.next();
+							while (i.hasNext()) {
+								final ListItemProxy item = i.next();
+								final KrollDict selectedItem = new KrollDict();
+								final ListSectionProxy section =
+									item.getParent() != null ? (ListSectionProxy) item.getParent() : null;
 
-						Log.d(TAG, "SELECTED: " + proxy.getProperties().getString(TiC.PROPERTY_TITLE));
+								if (section != null) {
+									selectedItem.put(TiC.PROPERTY_ITEM_INDEX, item.getIndexInSection());
+									selectedItem.put(TiC.PROPERTY_SECTION, section);
+									selectedItem.put(TiC.PROPERTY_SECTION_INDEX, proxy.getIndexOfSection(section));
+
+									selectedItems.add(selectedItem);
+								}
+							}
+
+							if (selectedItems.size() > 0) {
+								final KrollDict data = new KrollDict();
+
+								data.put(TiC.PROPERTY_SELECTED_ITEMS, selectedItems.toArray(new KrollDict[0]));
+								data.put(TiC.PROPERTY_STARTING_ITEM, selectedItems.get(0));
+								proxy.fireEvent(TiC.EVENT_ITEMS_SELECTED, data);
+							}
```

Comment:
We need to fire an "itemsselected" event when you uncheck all items too, where the selected items count is zero. So, we need to get rid of the `if (selectedItems.size() > 0) {}` part and do the following instead.
```java
final KrollDict data = new KrollDict();
data.put(TiC.PROPERTY_SELECTED_ITEMS, selectedItems.toArray(new KrollDict[0]));
data.put(TiC.PROPERTY_STARTING_ITEM, selectedItems.isEmpty() ? null : selectedItems.get(0));
proxy.fireEvent(TiC.EVENT_ITEMS_SELECTED, data);
```